dissect/util/stream.py

```diff
@@ -202,13 +265,29 @@ def __init__(self, fh: BinaryIO, offset: int, size: int, align: int = STREAM_BUF
         self._fh = fh
         self.offset = offset
 
+    def _seek(self, pos: int, whence: int = io.SEEK_SET) -> int:
+        if self.size is None and whence == io.SEEK_END:
+            pos = self._fh.seek(pos, whence)
+            if pos is None:
+                pos = self._fh.tell()
+            return max(0, pos - self.offset)
+        return super()._seek(pos, whence)
+
     def _read(self, offset: int, length: int) -> bytes:
-        read_length = min(length, self.size - offset)
+        # We will generally only end up here from :func:`AlignedStream.readall`
+        read_length = min(length, self.size - offset) if self.size else length
         self._fh.seek(self.offset + offset)
         return self._fh.read(read_length)
 
+    def _readinto(self, offset: int, buf: memoryview) -> int:
+        if not hasattr(self._fh, "readinto"):
```

Comment:
The other subclasses define a `_has_readinto` property, why not do the same here?